content/blog/tidymodels-2022-q4/index.Rmd

```diff
@@ -0,0 +1,262 @@
+categories: [roundup] 
+tags: [tidymodels, recipes, parsnip, rsample]
+---
+
+```{r}
+#| include: false
+#| label: startup
+
+library(tidymodels)
+
+tidymodels_prefer()
+theme_set(theme_bw())
+options(pillar.advice = FALSE, pillar.min_title_chars = Inf, width = 120)
+```
+
+```{r}
+#| label: get-repo-info
+#| include: FALSE
+
+since <- "2022-10-19"
+
+source("repo-functions.R")
+
+tm_data <- 
+  map_dfr(tm_pkgs, get_current_release) %>% 
+  filter(date > ymd(since)) %>% 
+  mutate(
+    repo = paste0("tidymodels/", package),
+    thanks = map_chr(repo, return_tidy_thanks, from = since),
+    thanks = glue("- {package}: {thanks}"),
+    news = glue("- {package} [({version})](https://{package}.tidymodels.org/news/index.html)")
+  )
+
+txt_pkg_list <- knitr::combine_words(tm_data$package)
+```
+
+The [tidymodels](https://www.tidymodels.org/) framework is a collection of R packages for modeling and machine learning using tidyverse principles.
+
+Since the beginning of 2021, we have been publishing [quarterly updates](https://www.tidyverse.org/categories/roundup/) here on the tidyverse blog summarizing what's new in the tidymodels ecosystem. The purpose of these regular posts is to share useful new features and any updates you may have missed. You can check out the [`tidymodels` tag](https://www.tidyverse.org/tags/tidymodels/) to find all tidymodels blog posts here, including our roundup posts as well as those that are more focused, like these posts from the past couple months:
+
+-   [tidyclust is on CRAN](https://www.tidyverse.org/blog/2022/12/tidyclust-0-1-0/)
+-   [Model calibration](https://www.tidyverse.org/blog/2022/11/model-calibration/)
+-   [Improvements to model specification checking in tidymodels](https://www.tidyverse.org/blog/2022/10/parsnip-checking-1-0-2/)
+
+Since [our last roundup post](https://www.tidyverse.org/blog/2022/10/tidymodels-2022-q3/), there have been CRAN releases of `r nrow(tm_data)` tidymodels packages. Here are links to their NEWS files:
+
+```{r}
+#| echo: FALSE
+#| results: asis
+
+cat(tm_data$news, sep = "\n")
+```
+
+We'll highlight a few especially notable changes below: more specialized role selectors in recipes, extended support for grouped resampling in rsample, and a big speedup in parsnip. First, loading the collection of packages:
+
+```{r}
+#| eval: FALSE
+library(tidymodels)
+```
+
+## Specialized role selectors
+
+The [recipes package for preprocessing](https://recipes.tidymodels.org/) supports tidyselect-style variable selection, and includes some of its own selectors to support common modeling workflows.
+
+To illustrate, we'll make use of a dataset `silly_data` with a number of different variable types:
+
+```{r}
+#| include: FALSE
+assign_letter <- function(names) {
+  c("class", letters[1:(length(names)-1)])
+}
+
+goofy_data <- 
+  sim_classification(num_linear = 3) %>%
+  mutate(across(ends_with("_1"), ~ as.factor(round(.x)))) %>%
+  mutate(two_factor_1 = as.integer(two_factor_1)) %>%
```

Comment:
I would turn more of the double variables into integers. It is hard to see the difference in the output when only `a` isn't affected

Reply:
I agree! We'll now create three rather than one.